cmd/vulcan-exposed-router-ports/main.go

```diff
@@ -61,44 +62,36 @@ var (
 	}
 )
 
-func exposedRouterPorts(target string, nmapReport *gonmap.NmapRun) []report.Vulnerability {
-	gr := report.ResourcesGroup{
-		Name: "Network Resources",
-		Header: []string{
-			"Hostname",
-			"Port",
-			"Protocol",
-			"Service",
-			"Version",
-		},
-	}
-
-	add := false
+func exposedRouterPorts(target string, nmapReport *gonmap.NmapRun, state checkstate.State) {
 	for _, host := range nmapReport.Hosts {
 		for _, port := range host.Ports {
 			if port.State.State != "open" {
 				continue
 			}
-
-			add = true
-
-			networkResource := map[string]string{
-				"Hostname": target,
-				"Port":     strconv.Itoa(port.PortId),
-				"Protocol": port.Protocol,
-				"Service":  port.Service.Product,
-				"Version":  port.Service.Version,
-			}
-			gr.Rows = append(gr.Rows, networkResource)
+			vuln := exposedVuln
+			vuln.AffectedResource = fmt.Sprintf("%d/%s", port.PortId, port.Protocol)
+			vuln.Fingerprint = helpers.ComputeFingerprint(port.Protocol, port.Service)
```

Comment:
I'd add the `Product` instead of the `Protocol`. In fact the second doesn't provide value to the fingerprint because when the protocol changes, the affected resource changes too, and therefore will be actually a different finding.

Reply:
Gotcha!